homeassistant/components/homewizard/sensor.py

```diff
@@ -452,3 +544,86 @@ def native_value(self) -> StateType:
         return super().available and self.native_value is not None
+
+
+class HomeWizardExternalSensorEntity(HomeWizardEntity, SensorEntity):
+    """Representation of externally connected HomeWizard Sensor."""
+
+    def __init__(
+        self,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        description: HomeWizardExternalSensorEntityDescription,
+        device_info: DeviceInfo,
+        device_unique_id: str,
+    ) -> None:
+        """Initialize Externally connected HomeWizard Sensors."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{DOMAIN}_{device_unique_id}"
+        self._device_id = device_unique_id
+        self._suggested_device_class = description.suggested_device_class
+        self._attr_device_info = device_info
+        self.entity_description = description
+
+    @property
+    def native_value(self) -> float | int | str | None:
+        """Return the sensor value."""
+        return self.device.value if self.device is not None else None
+
+    @property
+    def device(self) -> ExternalDevice | None:
+        """Return ExternalDevice object."""
+        return (
+            self.coordinator.data.data.external_devices[self._device_id]
+            if self.coordinator.data.data.external_devices is not None
+            else None
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return availability of meter."""
+        return super().available and self.device is not None
+
+    @property
+    def native_unit_of_measurement(self) -> str | None:
+        """Return unit of measurement based on device unit."""
+        if (device := self.device) is None:
+            return None
+
+        # API returns 'm3' but we expect m³
+        if device.unit == "m3":
+            return UnitOfVolume.CUBIC_METERS
+
+        return device.unit
+
+    @property
+    def device_class(self) -> SensorDeviceClass | None:
+        """Validate unit of measurement and set device class."""
+        if (
+            self.native_unit_of_measurement
+            not in DEVICE_CLASS_UNITS[self._suggested_device_class]
+        ):
+            return None
+
+        return self._suggested_device_class
+
+
+class HomeWizardExternalIdentifierSensorEntity(HomeWizardEntity, SensorEntity):
+    """Representation of externally connected HomeWizard Sensor."""
+
+    _attr_icon = "mdi:alphabetical-variant"
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_translation_key = "meter_identifier"
+
+    _attr_name = None
+
+    def __init__(
+        self,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        device_info: DeviceInfo,
+        device_unique_id: str,
+    ) -> None:
+        """Initialize Externally connected HomeWizard Sensors."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{DOMAIN}_{device_unique_id}_meter_identifier"
+        self._attr_native_value = device_unique_id
+        self._attr_device_info = device_info
```

Comment:
Instead of adding this sensor just to show the serial number, can you add it as serial number to the device?
https://github.com/home-assistant/core/blob/95fdae5e9d050e1a71781ad00a7a8b3046b1c0fd/homeassistant/helpers/device_registry.py#L93

Reply:
That is an option? cool!

I will take a quick look at that, as I think that would suggest that the P1 meter's serial number is that of the gas meter.